pkg/compute/kube/port_forward.go

```diff
@@ -30,12 +34,26 @@ type portForward struct {
 	Ready chan struct{}
 }
 
-func (ds DebugConfig) forwardPort(req portForward) {
+// PortForwardToProxy takes the given proxy and port forwards to the pod from this proxy
+func (c *client) PortForwardToProxy(proxy *models.Proxy, w http.ResponseWriter, url string, kubeConfig *rest.Config) {
```

Comment:
this api is not very flexible; for instance, it can't be used by a CLI since it requires writing to an http.ResponseWriter. How can we make this more flexible?

Also, the name `PortForwardToProxy` doesn't describe the inputs, outputs, but instead describes the internals of *how* it's achieving getting the config. I'd suggest changing that as well.

An alternative would be to either return the envoy config, or simply setup the port forwarding, and let the client call out to localhost.


Also notice that there is a port-forwarder already in `pkg/k8s/port-forward.go`, it likely makes sense to reuse this